src/ngx_http_lua_util.c

```diff
@@ -4093,4 +4093,33 @@ ngx_http_lua_cleanup_free(ngx_http_request_t *r, ngx_http_cleanup_pt *cleanup)
 }
 
 
+ngx_int_t
+ngx_http_lua_intercept_log_handler(ngx_log_t *log,
+    ngx_uint_t level, void *buf, size_t n)
+{
+    ngx_http_lua_main_conf_t      *lmcf;
+    ngx_http_lua_intercept_log_t  *intercept_log;
+
+    dd("enter");
+
+    lmcf = ngx_http_cycle_get_module_main_conf(ngx_cycle, ngx_http_lua_module);
+    if (lmcf == NULL) {
+        return NGX_DECLINED;
+    }
+
+    intercept_log = &lmcf->intercept_log;
+
+    if (n > intercept_log->max_size - intercept_log->size) {
+        return NGX_DECLINED;
```

Comment:
Instead of discarding the latest error log messages, we should discard the oldest ones in the buffer instead. New error log messages are almost always more valuable than the oldest ones. We need to make the buffer still get updated with the latest error logs even when the Lua application fails to extract the logs in time.